src/pages/index.js

```diff
@@ -4,73 +4,219 @@ import Section from "../components/Section.js";
+    })
+    .catch((err) => {
+      console.error(err);
+    });
 
-constants.addCardButton.addEventListener("click", () => {
-  addCardForm.open();
-});
+  function renderCard(cardData) {
+    const card = new Card(
+      cardData,
+      "#card-template",
+      handleImageClick,
+      handleCardDeleteClick,
+      handleLikeClick
+    ).getView();
+    cardSection.addItem(card);
+  }
 
-const cardSection = new Section(
-  {
-    renderer: (item) => {
-      const cardElement = createCard(item);
-      cardSection.addItem(cardElement);
-    },
-  },
-  constants.cardListEl
-);
+  const cardPreview = new PopupWithImage("#image-popup");
+  const confirmDeletePopup = new PopupWithConfirm("#delete-card-popup");
+  cardPreview.setEventListeners();
+
+  const profileEditValidator = new FormValidator(
+    constants.config,
+    document.querySelector("#profile-form")
+  );
+  const addCardValidator = new FormValidator(
+    constants.config,
+    document.querySelector("#add-card-form")
+  );
+  const avatarValidator = new FormValidator(
+    constants.config,
+    document.querySelector("#avatar-form")
+  );
+
+  profileEditValidator.enableValidation();
+  addCardValidator.enableValidation();
+  avatarValidator.enableValidation();
+
+  const avatarImagePopup = new PopupWithForm(
+    "#edit-avatar-popup",
+    handleAvatarFormSubmit
+  );
+
+  avatarImagePopup.setEventListeners();
+  document.querySelector(".profile__avatar").addEventListener("click", () => {
+    console.log("Avatar image clicked");
+    avatarImagePopup.open();
+  });
+
+  function handleImageClick(name, link) {
+    cardPreview.open({ name, link });
+  }
+
+  function createCard(cardData) {
+    const cardElement = new Card(
+      cardData,
+      "#card-template",
+      handleImageClick,
+      handleCardDeleteClick,
+      handleLikeClick
+    );
+    return cardElement.getView();
+  }
+
+  const profileEditForm = new PopupWithForm("#profile-edit-popup", (data) => {
+    const { header, description } = data;
+    profileEditForm.renderLoading(true);
 
-cardSection.renderItems(constants.initialCards);
+    api
+      .updateUserInfo(header, description)
+      .then((res) => {
+        userInformation.setUserInfo(res);
+        userInformation.setUserAvatar(res);
+        profileEditForm.close();
+        profileEditValidator.disableButton();
+      })
+      .catch((err) => {
+        console.error("Error updating profile:", err);
+      })
+      .finally(() => {
+        profileEditForm.renderLoading(false);
+      });
+  });
+
+  constants.profileEditButton.addEventListener("click", () => {
+    profileEditValidator.resetValidation();
+    api
+      .getUserInfo()
+      .then((userInfo) => {
+        constants.profileTitleInput.value = userInfo.name || "";
+        constants.profileDescriptionInput.value = userInfo.about || "";
+        profileEditForm.open();
+      })
+      .catch((err) => {
+        console.error("Error fetching user info:", err);
+      });
```

Comment:
Please, delete the duplicated code of profile form opening. You have a correct variant in line 186